cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/handlers/NGSIRestHandler.java

```diff
@@ -316,12 +324,31 @@ public void configure(Context context) {
         LOGGER.info("[NGSIRestHandler] Received data (" + data + ")");
         
         // Parse the original data into a NotifyContextRequest object
-        NotifyContextRequest ncr;
+        NotifyContextRequest ncr = null;
+        NotifyContextRequestNGSIv2 notifyContextRequestNGSIv2 = null;
         Gson gson = new Gson();
 
         try {
-            ncr = gson.fromJson(data, NotifyContextRequest.class);
-            LOGGER.debug("[NGSIRestHandler] Parsed NotifyContextRequest: " + ncr.toString());
+            if (ngsiVersion != null) {
+                switch (ngsiVersion) {
+                    case "legacy":
+                        ncr = gson.fromJson(data, NotifyContextRequest.class);
+                        LOGGER.debug("[NGSIRestHandler] Parsed NotifyContextRequest on legacy NGSI: " + ncr.toString());
+                        break;
+                    case "normalized":
+                        gson = new GsonBuilder().registerTypeAdapter(NotifyContextRequestNGSIv2.class, new NotifyContextRequestNGSIv2Deserializer()).create();
+                        notifyContextRequestNGSIv2 = gson.fromJson(data, NotifyContextRequestNGSIv2.class);
+                        ncr = notifyContextRequestNGSIv2.toNotifyContextRequest();
+                        LOGGER.debug("[NGSIRestHandler] Parsed NotifyContextRequest on normalized NGSIv2: " + ncr.toString());
+                        break;
+                    default:
```

Comment:
Is it compatible with NGSv1 notifications?

Reply:
Absolutely, if header `ngsiv2-attrsformat` is not provided (as a legacy mode) it processes request as NGSIv1. Also, if `ngsiv2-attrsformat` is provided as "legacy" is proccessed as NGSIv1.